### PR TITLE
Update Log Analytics Solutions Variable 

### PR DESCRIPTION
**Pull request by luke-taylor**
## Describe your changes
In order to allow maximum flexibility the variable as been updated to allow the user to specify both the product and the publisher thus eliminating the need for additional customisation.  

## Checklist before requesting a review
- [x] The pr title can be used to describe what this pr did in `CHANGELOG.md` file
- [x] I have executed pre-commit on my machine
- [x] I have passed pr-check on my machine

Thanks for your cooperation!

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -76,18 +76,18 @@ resource "azurerm_log_analytics_linked_service" "management" {
 }
 
 resource "azurerm_log_analytics_solution" "management" {
-  for_each = toset(var.log_analytics_solution_names)
+  for_each = { for plan in toset(var.log_analytics_solution_plans) : plan.product => plan }
```
Comment (matt-FFFFFF) [inline main.tf:79]: This is loads better I think, thanks!

Can we change the key to be "{plan.publisher}/{plan.product}"

Please? It just rules out any dupes

File: variables.tf [hunk 2]
```diff
@@ -66,23 +66,62 @@ variable "deploy_resource_group" {
   description = "A boolean flag to determine whether to deploy the Azure Resource Group or not."
 }
 
-variable "log_analytics_solution_names" {
-  type = list(string)
+variable "log_analytics_solution_plans" {
+  type = list(object({
+    product   = string
+    publisher = string
```
Comment (matt-FFFFFF) [inline variables.tf:72]: Maybe worth using optional(string, "Microsoft") here?

**Review by matt-FFFFFF — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (luke-taylor) [reply main.tf]: Thanks Matt, all done!

File: variables.tf (hunk 2 above)
Comment (luke-taylor) [reply variables.tf]: Yep that would be helpful, added now.

**Review by matt-FFFFFF — APPROVED** [review]

**Pull request merged**